### PR TITLE
Cash App (Bindings + PaymentSheet)

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- Adds Cash App Pay as a supported PM via bindings and within PaymentSheet
- Handles a new Cash App next action to redirect into a web view using follow redirects
- Adds some automated testing
- Adds some misc. formatting fixes required to pass CI

Live mode payment:
https://user-images.githubusercontent.com/88012362/212989708-75c2def6-9700-4a96-b16a-9cb997813b47.mov

## Motivation
https://docs.google.com/document/d/1j3adW1blskr4Yoki5zh-BrHxoP0GndvDQcq3hMrdBNA/edit?ouid=110487678322327336912&usp=docs_home&ths=true

## Testing
- Automated tests
- In test mode manulally
- In live mode manually

## Changelog
N/A, still in private beta

### CODE REVIEW

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PaymentSheetUITest: XCTestCase {
     var app: XCUIApplication!
 
     override func setUpWithError() throws {
+        super.setUpWithError()
```
Comment (wooj-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:15]: in xcode 13, it seems that this needs to be:
`try super.setUpWithError()`

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/CashAppMandateView.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
+//
+//  CashAppMandateView.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 1/10/23.
+//
+
+import Foundation
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/CashAppMandateView.swift:8]: Seems that `import Foundation` isn't needed here.

File: Stripe/StripeiOSTests/STPPaymentMethodCashAppTests.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            completion(cashAppJson ?? [:])
+        }
+    }
+
+    func testObjectDecoding() {
+        let retrieveJSON = XCTestExpectation(description: "Retrieve JSON")
+
+        _retrieveCashAppJSON({ json in
+            let cashApp = STPPaymentMethodCashApp.decodedObject(fromAPIResponse: json)
+            XCTAssertNotNil(cashApp, "Failed to decode JSON")
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply Stripe/StripeiOSTests/STPPaymentMethodCashAppTests.swift:33]: Good catch, I've updated the test!

File: StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodParams.swift [hunk 4]
```diff
@@ -514,6 +516,24 @@ public class STPPaymentMethodParams: NSObject, STPFormEncodable {
         self.metadata = metadata
     }
 
+    /// Creates params for an Cash App PaymentMethod. :nodoc:
```
[the remark this answers is not in the document]
Comment (porter-stripe) [reply StripePayments/StripePayments/Source/API Bindings/Models/PaymentMethods/STPPaymentMethodParams.swift]: Some copy pasta, not sure what it does but we have it on some of our other init's. Removed it.

**Review by wooj-stripe — APPROVED** [review]
lgtm!

**Pull request merged**